docs/monthday.md

```diff
@@ -248,3 +248,5 @@ md = Temporal.MonthDay.from('02-29');
 md.withYear(2017)  // throws
 md.withYear(2020)  // => 2020-02-29
 ```
+
+In calendars where more information than just the year is needed to convert a `Temporal.MonthDay` to a `Temporal.Date`, you can pass an object to `withYear()` that contains the necessary properties.
```

Comment:
It's worth giving an example here.

```javascript
let md = Temporal.MonthDay.from({
  calendar: "japanese",
  month: 1,
  day: 1
});

let date = md.withYear({ era: "reiwa", year: 2 });
```

Reply:
I tried to avoid doing that here, intending to postpone it until later, for two reasons: it won't currently work with the polyfill if someone tries it out, whereas all the rest of the examples do; and because of #526 the `era: 'reiwa'` is only going to work in the hacky version of the Japanese calendar implementation in my branch, and we may have to pick a different way to specify eras in the end.

I'll make sure to add this example to the calendar branch though, so that we make sure it gets merged into here in whatever form it ultimately takes.

Reply:
OK.